transmission_interface/test/simple_transmission_loader_test.cpp

```diff
@@ -0,0 +1,382 @@
+// Copyright 2022 PAL Robotics S.L.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "transmission_interface/simple_transmission_loader.hpp"
+#include <pluginlib/class_loader.hpp>
+
+#include <gtest/gtest.h>
+#include <exception>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "transmission_interface/simple_transmission.hpp"
+#include "transmission_interface/transmission_loader.hpp"
+
+class TransmissionPluginLoader
+{
+public:
+  std::shared_ptr<transmission_interface::TransmissionLoader> create(const std::string & type)
+  {
+    try
+    {
+      return class_loader_.createUniqueInstance(type);
+    }
+    catch (std::exception & ex)
+    {
+      std::cerr << ex.what() << std::endl;
+
+      return std::shared_ptr<transmission_interface::TransmissionLoader>();
+    }
+  }
+
+private:
+  // must keep it alive because instance destroyers need it
+  pluginlib::ClassLoader<transmission_interface::TransmissionLoader> class_loader_ = {
+    "transmission_interface", "transmission_interface::TransmissionLoader"};
+};
+
+TEST(SimpleTransmissionLoaderTest, FullSpec)
+{
+  // Parse transmission info
+
+  std::string urdf_to_test =
```

Comment:
Why not to add this into `ros2_control_test_assets`? It would be a better and reusable place for this. Maybe there is already exactly the one you need. Or perhaps we can only extend the existing `MinimalRobot` with this.

Reply:
Let's do a follow-up of this please. I don't think that adding everything to test assets it a good idea, it will just become a dumpster for test stuff which we also don't want. It is supposed to store _reusable_ test assets.

Reply:
That's my points. I think this urdf for test is reusable. We should extend the main URDF for tests and add all features in there. It doesn't matter is some of those are not used in all testing scenarios. But they have to anyway be parsable and do not collide with other features. I will open follow-up issue.